docs/core/deploying/native-aot.md

```diff
@@ -0,0 +1,70 @@
+---
+title: NativeAOT deployment overview
+description: Learn what NativeAOT deployments are and why you should consider using it as part of the publishing your app with .NET 7 and later.
+author: lakshanf
+ms.author: lakshanf
+ms.date: 06/09/2022
+---
+# NativeAOT Deployment
+
+The .NET NativeAOT deployment model uses an optimized ahead-of-time (AOT) compiler that generates an application that contains native executable code. NativeAOT apps don't use the just-in-time (JIT) compiler when the application runs. Native AOT provides performance benefits (for example, startup time and working set improvements) and the ability to run .NET applications in restricted environments where a JIT is not allowed. NativeAOT applications are [self-contained applications](index.md#publish-self-contained) that target a specific runtime environment (RID), such as Linux x64 or Windows x64.
+
+There are some limitations in the .NET NativeAOT deployment model, with the main one being that no run-time code generation is possible. For more information, see [Impact of using the NativeAOT deployment](#impact-of-using-the-nativeaot-deployment). The support in the .NET 7 release is targeted towards console-type applications.
+
+> [!WARNING]
+> NativeAOT deployment is an experimental feature in .NET 7.
+
+## Publish NativeAOT - CLI
+
+Publish a NativeAOT application using the [dotnet publish](../tools/dotnet-publish.md) command.
+
+01. Add `<PublishAot>true</PublishAot>` to your project file.
+
+    This will produce a NativeAOT app and shows PublishAot compatibility warnings during build.
+
+    ```xml
+    <PropertyGroup>
+        <PublishAot>true</PublishAot>
+    </PropertyGroup>
+    ```
+
+01. Publish the app as for a specific runtime identifier using `dotnet publish -r <RID>`
+
+    The following example publishes the app for Windows as a NativeAOT application.
+
+    `dotnet publish -r win-x64 -c Release`
+
+    The following example publishes the app for Linux as a NativeAOT application.
+
+    `dotnet publish -r linux-arm64 -c Release`
+
+The app will be available in the publish directory and will contain all the code needed to run in it, including the coreclr runtime.
+
+NativeAOT libraries that can be consumed by other programming languages can be published using the `NativeLib=Static` property for static libraries that can be linked at compile time. Shared libraries that are required at runtime can be created at publish time by using the `NativeLib=Shared` property.
```

Comment:
Based on my experience Shared library mode is break implicit expectations of dev community. A lot of developers would not realize that their native code would have CLR running inside their app and second DLL compiled with NativeAOT in same process would be problematic. That should be highlighted since a lot of people want develop in C# and use artifacts in unmanaged world.

Reply:
> and second DLL compiled with NativeAOT in same process would be problematic

What kind of problems did you run into with that setup?

It's not expected to be more problematic than any other "there are two automatic memory managers in the process that don't know about each other" situation. The other library doesn't have to be NativeAOT-compiled.

NativeAOT doesn't have the problem as CoreCLR that there can only be one of those in the process.

Reply:
I have only limited exposure to this setup and in very ìnteresting` circumstances. I try to create custom GC and it chokes because some resources was shared between NativeAOT GC initialization phase and CoreCLR Gc initialization phase. If you can have 2 runtimes and 2 GC within same process then I my remark would not be applicable. I may misinterpret things.

Reply:
This is getting really off topic but if you're referring to this setup: https://twitter.com/KooKiz/status/1533133858312445956, the problem there is about trying to do `[UnmanagedCallersOnly(Entrypoint="GC_Initialize")]`. It won't link because Native AOT uses that symbol name. One can run into the same problem with `[UnmanagedCallersOnly(Entrypoint="printf")]`. The solution is: "don't do that". The symbolic name is gone after the linker runs and doesn't pose problems for a "finished" library.

Reply:
Yeah. You are right. I did try something like what's in that tweet about year ago, but did not manage to finish experiment  properly.

Reply:
Removed references to advanced scenarios